specs/design-black-box.md

```diff
@@ -0,0 +1,58 @@
+# GitOps Engine Design - Black Box
+
+## Summary
+
+During the elaboration of [White box](./design-white-box.md) option, it was discovered that some components are similar at a high-level but still have a lot of differences in 
+design and implementation. This is not surprising because code was developed by different teams and with a focus on different use-cases. Given that it would require a lot of
+effort to resolve such differences it is proposed contributing missing features of one project into an engine of another and use that engine in both projects.
+
+## Proposal
+
+It is proposed to use Argo CD application controller as the base for the GitOps engine and contribute a set of Flux features into it. There are two main reasons to try using Argo CD as
+a base:
+- Argo CD uses Application abstraction to represent the desired state and target Kubernetes cluster. This abstraction works for both Argo CD and Flux cases.
+- The Argo CD controller leverages Kubernetes watch APIs instead of pulling. This enables Argo CD features such as Health assessment, UI and could provide better performance to
```

Comment:
```suggestion
- The Argo CD controller leverages Kubernetes watch APIs instead of polling. This enables Argo CD features such as Health assessment, UI and could provide better performance to
```